cli/internal/cmd/create.go

```diff
@@ -202,6 +202,9 @@ func (c *createCmd) create(cmd *cobra.Command, applier cloudApplier, fileHandler
 	if err != nil {
 		return fmt.Errorf("reading state file: %w", err)
 	}
+	if err := stateFile.Validate(state.PreCreate, provider); err != nil {
+		return fmt.Errorf("validating state file: %w", err)
+	}
```

Comment:
Here we probably want to run validation at the start of the `create` function to prevent creating a cluster only to then fail after everything was created by Terraform because the state file wasn't empty.
This does not matter too much, since this function will be removed/reworked in https://github.com/edgelesssys/constellation/pull/2549